crates/host/src/wasmbus/mod.rs

```diff
@@ -274,17 +277,24 @@ impl wrpc_transport::Serve for WrpcServer {
             let metrics = Arc::clone(&metrics);
             let policy_manager = Arc::clone(&policy_manager);
             let trace_ctx = Arc::clone(&trace_ctx);
-            // NOTE(thomastaylor312): We create a span each time here for two reasons: First
-            // off, if we create a separate span and then instrument this whole block of code,
-            // it makes it so the function isn't FnMut. So we create this each time. The second
-            // reason is that before we were trying to use the current span for instrumentation.
-            // This didn't work because the span created by the `instrument` macro exits after
-            // this function returns, so we ended up with no parent span at all when trying to
-            // attach header data. I also set this as an info span so that each invocation at
-            // least has this top level span. If the fields add too much overhead, we can remove
-            // those fields instead.
             let span = tracing::info_span!("component_invocation", func = %func, id = %id, instance = %instance);
             async move {
+                if let Some(ref cx) = cx {
+                    // TODO: wasmcloud_tracing take HeaderMap for my own sanity
```

Comment:
Do we need/want to capture a follow-up for this?

Reply:
Nobody seems to know what this means. I have moved this from original implementation in one of the refactors.
If we cannot decipher this comment, maybe we should just remove it finally

Reply:
I'm 👍 for removing if we can't think of a reason why it should continue to exist.

Reply:
I wrote it 😢 and it was just a comment to avoid the random `iter()` here. I think we can remove it